### PR TITLE
Feature/73 test comments endpoints

### PR DESCRIPTION
**Pull request by daviskoh**
Changed:

- add tests for used /comments endpoints
- update those endpoints to use async/await
- add fixtures for re-use

### CODE REVIEW

File: server/routes/comments.js [hunk 1]
```diff
@@ -55,25 +56,18 @@ function validateBody(req, res, next) {
 class Comments {
   constructor() {
     this.router = express.Router({ mergeParams: true });
-    this.router.get('/', this.getAll);
+    this.router.get('/', wrap(this.getAll));
     this.router.get('/:commentId', this.getById);
```
Comment (jackofseattle) [inline server/routes/comments.js]: Why wouldn't getById need to be wrapped?

**Comment by jackofseattle** [comment]
LGTM - would be nice if we could fix the issue with the fixture before moving forward though.

File: server/routes/comments.js (hunk 1 above)
Comment (daviskoh) [inline server/routes/comments.js]: I didnt wrap or test it because I dont think we are using it to my knowledge. @jshor is the handler ok to remove?

**Comment by daviskoh** [comment]
Currently looking into the issue :sake:

File: server/routes/comments.js (hunk 1 above)
Comment (jshor) [inline server/routes/comments.js]: @daviskoh we can remove it

**Comment by jshor** [comment]
merge conflicts, otherwise LGTM

**Comment by daviskoh** [comment]
:sob:

File: test/server/candidates.js [hunk 2]
```diff
@@ -16,34 +16,17 @@ let mockPut;
...
-        "shortDescription": "Ut velit natus vel et itaque laboriosam qui est quia.",
-        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/dawidwu/128.jpg"
-      }
-    ];
 
     mockGetAll = candidateArray;
-    mockGet = candidateArray[0];
-    mockPost = candidateArray[0];
-    mockPut = candidateArray[1];
+    mockGet = Object.assign({}, candidateArray[0]);
```
Comment (jackofseattle) [inline test/server/candidates.js]: Seems like you could just do object assign when you create the candidateArray and now have to do it on every line here.

**Comment by jackofseattle** [comment]
LGTM

**Pull request merged**